s3_prototype/main.tf

```diff
@@ -0,0 +1,82 @@
+resource "aws_s3_bucket" "bucket" {
+  bucket = var.bucket_name
+  tags = merge(
+    var.common_tags,
+    tomap(
+      { "Name" = var.bucket_name }
+    )
+  )
+}
+
+resource "aws_s3_bucket_public_access_block" "bucket_public_access" {
+  bucket                  = aws_s3_bucket.bucket.id
+  block_public_acls       = true
+  block_public_policy     = true
+  ignore_public_acls      = true
+  restrict_public_buckets = true
+}
+
+resource "aws_s3_bucket_server_side_encryption_configuration" "encryption_configuration" {
+  bucket = aws_s3_bucket.bucket.id
+  rule {
+    apply_server_side_encryption_by_default {
+      sse_algorithm     = var.kms_key_arn == null ? "AES256" : "aws:kms"
+      kms_master_key_id = var.kms_key_arn
+    }
+    bucket_key_enabled = true
+  }
+}
+
+resource "aws_s3_bucket_versioning" "versioning" {
+  bucket = aws_s3_bucket.bucket.id
+  versioning_configuration {
+    status = "Enabled"
+  }
+}
+
+resource "aws_s3_bucket_lifecycle_configuration" "delete_incomplete_multipart_uploads" {
```

Comment:
This adds the lifecycle policy to the logging buckets as well. Until we get the logs replicating into another account then we might want to keep logs in the logging buckets longer. Is it worth adding a variable which determines whether to add this or not?

Reply:
ah thats a good point, I will add a variable to toggle this, with it off by default.
@kurtismash thoughts?

Reply:
actually on second thoughts, this lifecycle policy just about incomplete uploads so I think it is fine to keep for log buckets

Reply:
There's a thread on nationalarchives/ds-infrastructure-audit#27 about lifecycle policies on logging buckets.